src/secret.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from typing import Optional, ValuesView
+
+from ops import Model, SecretNotFoundError
+
+from configs import ServiceConfigs
+from constants import (
+    COOKIE_SECRET_KEY,
+    COOKIE_SECRET_LABEL,
+    SYSTEM_SECRET_KEY,
+    SYSTEM_SECRET_LABEL,
+)
+
+
+class Secrets:
```

Comment:
The `Secrets` class manages all of the juju secrets created and maintained by the charm itself.